src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,95 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.HashMap;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String STRING_SEPARATOR = ",";
+    private static final String[] REPORT_COLUMNS = new String[] {"supply", "buy", "result"};
+    private static BufferedWriter bufferedWriter = null;
+    private static File file = null;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        HashMap<String, Integer> results = calculateStatisticResults(fromFileName);
+        clearFile(toFileName);
+        file = new File(toFileName);
```

Comment:
You need to split the logic into three private methods and call them one by one in the public method.
1) only reads and returns rows
2) accepts rows calculates and generates a report
3) only write the report to a file